src/scenes/Auth/Auth.tsx

```diff
@@ -5,23 +5,19 @@ import { RouteComponentProps, Redirect } from '@reach/router';
 import useIsPasswordValidated from 'hooks/useIsPasswordValidated';
 import WalletSelector from 'components/WalletSelector/WalletSelector';
 import useIsAuthenticated from 'contexts/auth/useIsAuthenticated';
+import { useAuthenticatedUser } from 'hooks/api/useUser';
 
 function Auth(_: RouteComponentProps) {
   // whether the user has entered the correct password
   const isPasswordValidated = useIsPasswordValidated();
   // whether the user is web3-authenticated
   const isAuthenticated = useIsAuthenticated();
-  // TODO: we need a third hook, `useUser()`, that will determine if the
-  // user exists in the database so we can do something like below:
-  // const user = useUser();
-  // const userExists = !!user;
-  // const username = user?.username;
-  const userExists = false;
-  const username = '0x70d04384b5c3a466ec4d8cfb8213efc31c6a9d15';
+  const user = useAuthenticatedUser();
+  const userExists = !!user;
 
   if (isAuthenticated) {
     // if user exists in DB, send them to their profile
-    if (userExists) return <Redirect to={`/${username}`} />;
+    if (userExists) return <Redirect to={`/${user?.username}`} />;
```

Comment:
we should just be able to do `user.username` here because we ensured `user` exists

Reply:
Compiler doesn't like it :( "Object is possibly 'null'."
I could use `user!.username` instead, either seems ok for here